storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3Storage.java

```diff
@@ -54,12 +54,13 @@ public void configure(final Map<String, ?> configs) {
 
     @Override
     public long upload(final InputStream inputStream, final ObjectKey key) throws StorageBackendException {
-        try (final var out = s3OutputStream(key)) {
+        final var out = s3OutputStream(key);
+        try (out) {
             inputStream.transferTo(out);
-            return out.processedBytes();
         } catch (final IOException e) {
             throw new StorageBackendException("Failed to upload " + key, e);
         }
+        return out.processedBytes();
```

Comment:
Added a comment to clarify a bit, but the main reason this is now outside the resource try-catch is because the `processedBytes` in the output stream is counted now as part of the `flushBuffer` and not the `write` method anymore. 
This is more accurate as it counts bytes processed _after upload_ compared to the previous behavior where it was count when writing.
Before, the processed bytes would be known before closing. Now as the last `flushBuffer` may occur when closing the stream, it means that the result is not known before closing, thus the change.